fix(organizations): Update overview callout to point to orgs repo (#2252)
clerk/clerk-docs: docs/organizations/overview.mdx

Organizations are a flexible and scalable way to manage users and their access to resources within your Clerk application. With organizations, you can assign specific roles and permissions to users, making them useful for managing projects, coordinating teams, or facilitating partnerships.

> [!NOTE]

> > To explore organizations in Clerk, refer to our demo apps:

Reviewer: ```suggestion
> To explore organizations in Clerk, check out the demo apps:
```
